core/src/window_service.rs

```diff
@@ -218,38 +217,6 @@ fn run_check_duplicate(
         .try_for_each(check_duplicate)
 }
 
-fn verify_repair(
-    outstanding_requests: &mut OutstandingShredRepairs,
-    shred: &Shred,
-    repair_meta: &Option<RepairMeta>,
-) -> bool {
-    repair_meta
-        .as_ref()
-        .map(|repair_meta| {
-            outstanding_requests
-                .register_response(
-                    repair_meta.nonce,
-                    shred,
-                    solana_sdk::timing::timestamp(),
-                    |_| (),
-                )
-                .is_some()
-        })
-        .unwrap_or(true)
-}
-
-fn prune_shreds_by_repair_status(
-    shreds: &mut Vec<(Shred, Option<RepairMeta>)>,
-    outstanding_requests: &RwLock<OutstandingShredRepairs>,
-    accept_repairs_only: bool,
-) {
-    let mut outstanding_requests = outstanding_requests.write().unwrap();
```

Comment:
Another win looks to be that we were holding this lock while we iterate over all shreds. Now, we only hold the lock in the more granular context of scanning over ONLY repair shreds

Reply:
right